src/features/Movies/component.js

```diff
@@ -1,142 +1,123 @@
 /* eslint-disable max-len */
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
+import { ProjectContext } from '../../store/ProjectContext';
+// import PropTypes from 'prop-types';
 // import components
 import MovieCategories from '../MovieCategories';
 import MovieItem from '../MovieItem';
 import MovieFilters from '../MovieFilters';
 import AddMovie from '../../shared/AddMovie';
-import {filterArray} from '../../assets/functions/util';
-// import mockData 
-import { mockData } from '../../assets/data/MockData';
+import { filterArray } from '../../assets/functions/util';
 import { v4 as uuidv4 } from 'uuid';
 import './style.scss';
 
-export default class Movies extends Component {
-    // define our state
-    constructor(props){
-        super(props);
-        this.state = {
-			movieItem : {name : '', date : '', category : '', rating : '', img : '', overview : '', id : ''},
-            movies       : mockData,
-			moviesToShow : mockData,
-			categories   : [
-                { name: 'ALL', active: true },
-                { name: 'DOCUMENTARY', active: false },
-                { name: 'COMEDY', active: false },
-                { name: 'HORROR', active: false },
-                { name: 'ACTION', active: false }
-            ],
-			filter : 'RELEASE DATE'
-        };
-		this.handleChangeCategory = this.handleChangeCategory.bind(this);
-		this.handleChangeFilter = this.handleChangeFilter.bind(this);
-		this.handleCancelAddMovie = this.handleCancelAddMovie.bind(this);
-		this.handleChangeMovieDetails = this.handleChangeMovieDetails.bind(this);
-		this.handleSubmitMovie = this.handleSubmitMovie.bind(this);
-		this.handleResetMovie = this.handleResetMovie.bind(this);
-		this.handleEditMovie = this.handleEditMovie.bind(this);
-		this.handleRemoveMovie = this.handleRemoveMovie.bind(this);
+const Movies = () => {
+    // bind our Context api state with our component
+    const [
+        isHidden, setIsHidden, 
+        showMovie, setShowMovie,
+        // eslint-disable-next-line no-unused-vars
+        movieDetails, setMovieDetails,
+        movieItem, setMovieItem,
+        // eslint-disable-next-line no-unused-vars
+        movies, setMovies,
+        moviesToShow, setMoviesToShow,
+        categories, setCategories,
+        filter, setFilter
+    ] = useContext(ProjectContext);
+
+    // define our event handlers
+    const showMovieDetailsHandler = (movieId) => {
+        setShowMovie(!showMovie);
+        const movieDetailsUpdated = moviesToShow.filter((item) => item.id === movieId);
```

Comment:
useMemo

Reply:
I am not sure how to implement useMemo here?

Reply:
```suggestion
        const movieDetailsUpdated = useMemo(() => moviesToShow.filter((item) => item.id === movieId), [moviesToShow]);
```

Reply:
yeah, my bad